tox.ini

```diff
@@ -0,0 +1,17 @@
+[tox]
+envlist = ui-tests, flake8
+skipsdist = True
+
+[testenv:ui-tests]
+passenv = DISPLAY
+deps = -rfrontend/test/ui/requirements/requirements.txt
+commands =
+    pytest -v --driver=Firefox --firefox-path=firefox/firefox frontend/test/ui {posargs}
```

Comment:
If you add pytest-html and `--html` you can generate a HTML report. CircleCI can then archive this as an artifact, which may help with investigating issues.